{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc746f99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MMseqs2 (Many against Many sequence searching) is an open-source software suite for very fast, \r\n",
      "parallelized protein sequence searches and clustering of huge protein sequence data sets.\r\n",
      "\r\n",
      "Please cite: M. Steinegger and J. Soding. MMseqs2 enables sensitive protein sequence searching for the analysis of massive data sets. Nature Biotechnology, doi:10.1038/nbt.3988 (2017).\r\n",
      "\r\n",
      "MMseqs2 Version: 15-6f452+ds-2\r\n",
      "© Martin Steinegger (martin.steinegger@snu.ac.kr)\r\n",
      "\r\n",
      "usage: mmseqs <command> [<args>]\r\n",
      "\r\n",
      "Easy workflows for plain text input/output\r\n",
      "  easy-search       \tSensitive homology search\r\n",
      "  easy-linsearch    \tFast, less sensitive homology search\r\n",
      "  easy-cluster      \tSlower, sensitive clustering\r\n",
      "  easy-linclust     \tFast linear time cluster, less sensitive clustering\r\n",
      "  easy-taxonomy     \tTaxonomic classification\r\n",
      "  easy-rbh          \tFind reciprocal best hit\r\n",
      "\r\n",
      "Main workflows for database input/output\r\n",
      "  search            \tSensitive homology search\r\n",
      "  linsearch         \tFast, less sensitive homology search\r\n",
      "  map               \tMap nearly identical sequences\r\n",
      "  rbh               \tReciprocal best hit search\r\n",
      "  linclust          \tFast, less sensitive clustering\r\n",
      "  cluster           \tSlower, sensitive clustering\r\n",
      "  clusterupdate     \tUpdate previous clustering with new sequences\r\n",
      "  taxonomy          \tTaxonomic classification\r\n",
      "\r\n",
      "Input database creation\r\n",
      "  databases         \tList and download databases\r\n",
      "  createdb          \tConvert FASTA/Q file(s) to a sequence DB\r\n",
      "  createindex       \tStore precomputed index on disk to reduce search overhead\r\n",
      "  createlinindex    \tCreate linsearch index\r\n",
      "  convertmsa        \tConvert Stockholm/PFAM MSA file to a MSA DB\r\n",
      "  tsv2db            \tConvert a TSV file to any DB\r\n",
      "  tar2db            \tConvert content of tar archives to any DB\r\n",
      "  db2tar            \tArchive contents of a DB to a tar archive\r\n",
      "  msa2profile       \tConvert a MSA DB to a profile DB\r\n",
      "\r\n",
      "Handle databases on storage and memory\r\n",
      "  compress          \tCompress DB entries\r\n",
      "  decompress        \tDecompress DB entries\r\n",
      "  rmdb              \tRemove a DB\r\n",
      "  mvdb              \tMove a DB\r\n",
      "  cpdb              \tCopy a DB\r\n",
      "  lndb              \tSymlink a DB\r\n",
      "  aliasdb           \tCreate relative symlink of DB to another name in the same folder\r\n",
      "  unpackdb          \tUnpack a DB into separate files\r\n",
      "  touchdb           \tPreload DB into memory (page cache)\r\n",
      "\r\n",
      "Unite and intersect databases\r\n",
      "  createsubdb       \tCreate a subset of a DB from list of DB keys\r\n",
      "  concatdbs         \tConcatenate two DBs, giving new IDs to entries from 2nd DB\r\n",
      "  splitdb           \tSplit DB into subsets\r\n",
      "  mergedbs          \tMerge entries from multiple DBs\r\n",
      "  subtractdbs       \tRemove all entries from first DB occurring in second DB by key\r\n",
      "\r\n",
      "Format conversion for downstream processing\r\n",
      "  convertalis       \tConvert alignment DB to BLAST-tab, SAM or custom format\r\n",
      "  createtsv         \tConvert result DB to tab-separated flat file\r\n",
      "  convert2fasta     \tConvert sequence DB to FASTA format\r\n",
      "  result2flat       \tCreate flat file by adding FASTA headers to DB entries\r\n",
      "  createseqfiledb   \tCreate a DB of unaligned FASTA entries\r\n",
      "  taxonomyreport    \tCreate a taxonomy report in Kraken or Krona format\r\n",
      "\r\n",
      "Sequence manipulation/transformation\r\n",
      "  extractorfs       \tSix-frame extraction of open reading frames\r\n",
      "  extractframes     \tExtract frames from a nucleotide sequence DB\r\n",
      "  orftocontig       \tWrite ORF locations in alignment format\r\n",
      "  reverseseq        \tReverse (without complement) sequences\r\n",
      "  translatenucs     \tTranslate nucleotides to proteins\r\n",
      "  translateaa       \tTranslate proteins to lexicographically lowest codons\r\n",
      "  splitsequence     \tSplit sequences by length\r\n",
      "  masksequence      \tSoft mask sequence DB using tantan\r\n",
      "  extractalignedregion\tExtract aligned sequence region from query\r\n",
      "\r\n",
      "Result manipulation \r\n",
      "  swapresults       \tTranspose prefilter/alignment DB\r\n",
      "  result2rbh        \tFilter a merged result DB to retain only reciprocal best hits\r\n",
      "  result2msa        \tCompute MSA DB from a result DB\r\n",
      "  result2dnamsa     \tCompute MSA DB with out insertions in the query for DNA sequences\r\n",
      "  result2stats      \tCompute statistics for each entry in a DB\r\n",
      "  filterresult      \tPairwise alignment result filter\r\n",
      "  offsetalignment   \tOffset alignment by ORF start position\r\n",
      "  proteinaln2nucl   \tTransform protein alignments to nucleotide alignments\r\n",
      "  result2repseq     \tGet representative sequences from result DB\r\n",
      "  sortresult        \tSort a result DB in the same order as the prefilter or align module\r\n",
      "  summarizealis     \tSummarize alignment result to one row (uniq. cov., cov., avg. seq. id.)\r\n",
      "  summarizeresult   \tExtract annotations from alignment DB\r\n",
      "\r\n",
      "Taxonomy assignment \r\n",
      "  createtaxdb       \tAdd taxonomic labels to sequence DB\r\n",
      "  createbintaxonomy \tCreate binary taxonomy from NCBI input\r\n",
      "  createbintaxmapping\tCreate binary taxonomy mapping from tabular taxonomy mapping\r\n",
      "  addtaxonomy       \tAdd taxonomic labels to result DB\r\n",
      "  taxonomyreport    \tCreate a taxonomy report in Kraken or Krona format\r\n",
      "  filtertaxdb       \tFilter taxonomy result database\r\n",
      "  filtertaxseqdb    \tFilter taxonomy sequence database\r\n",
      "  aggregatetax      \tAggregate multiple taxon labels to a single label\r\n",
      "  aggregatetaxweights\tAggregate multiple taxon labels to a single label\r\n",
      "  lcaalign          \tEfficient gapped alignment for lca computation\r\n",
      "  lca               \tCompute the lowest common ancestor\r\n",
      "  majoritylca       \tCompute the lowest common ancestor using majority voting\r\n",
      "\r\n",
      "Multi-hit search    \r\n",
      "  multihitdb        \tCreate sequence DB for multi hit searches\r\n",
      "  multihitsearch    \tSearch with a grouped set of sequences against another grouped set\r\n",
      "  besthitperset     \tFor each set of sequences compute the best element and update p-value\r\n",
      "  combinepvalperset \tFor each set compute the combined p-value\r\n",
      "  mergeresultsbyset \tMerge results from multiple ORFs back to their respective contig\r\n",
      "\r\n",
      "Prefiltering        \r\n",
      "  prefilter         \tDouble consecutive diagonal k-mer search\r\n",
      "  ungappedprefilter \tOptimal diagonal score search\r\n",
      "  gappedprefilter   \tOptimal Smith-Waterman-based prefiltering (slow)\r\n",
      "  kmermatcher       \tFind bottom-m-hashed k-mer matches within sequence DB\r\n",
      "  kmersearch        \tFind bottom-m-hashed k-mer matches between target and query DB\r\n",
      "\r\n",
      "Alignment           \r\n",
      "  align             \tOptimal gapped local alignment\r\n",
      "  alignall          \tWithin-result all-vs-all gapped local alignment\r\n",
      "  transitivealign   \tTransfer alignments via transitivity\r\n",
      "  rescorediagonal   \tCompute sequence identity for diagonal\r\n",
      "  alignbykmer       \tHeuristic gapped local k-mer based alignment\r\n",
      "\r\n",
      "Clustering          \r\n",
      "  clust             \tCluster result by Set-Cover/Connected-Component/Greedy-Incremental\r\n",
      "  clusthash         \tHash-based clustering of equal length sequences\r\n",
      "  mergeclusters     \tMerge multiple cascaded clustering steps\r\n",
      "\r\n",
      "Profile databases   \r\n",
      "  result2profile    \tCompute profile DB from a result DB\r\n",
      "  msa2result        \tConvert a MSA DB to a profile DB\r\n",
      "  msa2profile       \tConvert a MSA DB to a profile DB\r\n",
      "  sequence2profile  \tTurn sequence into profile by adding context specific pseudo counts\r\n",
      "  profile2pssm      \tConvert a profile DB to a tab-separated PSSM file\r\n",
      "  profile2neff      \tConvert a profile DB to a tab-separated list of Neff scores\r\n",
      "  profile2consensus \tExtract consensus sequence DB from a profile DB\r\n",
      "  profile2repseq    \tExtract representative sequence DB from a profile DB\r\n",
      "  convertprofiledb  \tConvert a HH-suite HHM DB to a profile DB\r\n",
      "\r\n",
      "Profile-profile databases\r\n",
      "  tsv2exprofiledb   \tCreate a expandable profile db from TSV files\r\n",
      "  convertca3m       \tConvert a cA3M DB to a result DB\r\n",
      "  expandaln         \tExpand an alignment result based on another\r\n",
      "  expand2profile    \tExpand an alignment result based on another and create a profile\r\n",
      "\r\n",
      "Utility modules to manipulate DBs\r\n",
      "  setextendeddbtype \tWrite an extended DB \r\n",
      "  view              \tPrint DB entries given in --id-list to stdout\r\n",
      "  apply             \tExecute given program on each DB entry\r\n",
      "  filterdb          \tDB filtering by given conditions\r\n",
      "  swapdb            \tTranspose DB with integer values in first column\r\n",
      "  prefixid          \tFor each entry in a DB prepend the entry key to the entry itself\r\n",
      "  suffixid          \tFor each entry in a DB append the entry key to the entry itself\r\n",
      "  renamedbkeys      \tCreate a new DB with original keys renamed\r\n",
      "\r\n",
      "Special-purpose utilities\r\n",
      "  diffseqdbs        \tCompute diff of two sequence DBs\r\n",
      "  summarizetabs     \tExtract annotations from HHblits BLAST-tab-formatted results\r\n",
      "  gff2db            \tExtract regions from a sequence database based on a GFF3 file\r\n",
      "  maskbygff         \tMask out sequence regions in a sequence DB by features selected from a GFF3 file\r\n",
      "  convertkb         \tConvert UniProtKB data to a DB\r\n",
      "  summarizeheaders  \tSummarize FASTA headers of result DB\r\n",
      "  nrtotaxmapping    \tCreate taxonomy mapping for NR database\r\n",
      "  extractdomains    \tExtract highest scoring alignment regions for each sequence from BLAST-tab file\r\n",
      "  countkmer         \tCount k-mers\r\n",
      "\r\n",
      "Bash completion for modules and parameters can be installed by adding \"source MMSEQS_HOME/util/bash-completion.sh\" to your \"$HOME/.bash_profile\".\r\n",
      "Include the location of the MMseqs2 binary in your \"$PATH\" environment variable.\r\n",
      "\r\n",
      "Invalid Command: --version\r\n",
      "\u001b[33mDid you mean \"/usr/bin/mmseqs-avx2 reverseseq\"?\r\n",
      "\u001b[39m"
     ]
    }
   ],
   "source": [
    "#als erstes Kernel für Python in WSL auswählen\n",
    "!mmseqs --version\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c36cfbd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
